From 807fd2d518ea43c623268edecdc0b58361188af8 Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Wed, 20 Oct 2021 09:49:28 +0000
Subject: [PATCH 56/78] mlxsw: Use weak reverse dependencies for firmware
 flashing selection

Replace configuration options "select" by "imply". It allows to set
'MLXFW' symbol to 'n' from a direct dependency or with a visible
prompt.

User might wish to disable this option in case only 'mlxsw_minimal' is
configured, since it does not support firmware flashing.

Signed-off-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/net/ethernet/mellanox/mlxsw/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/Kconfig b/drivers/net/ethernet/mellanox/mlxsw/Kconfig
index 74f7e27..ae11732 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/Kconfig
+++ b/drivers/net/ethernet/mellanox/mlxsw/Kconfig
@@ -6,7 +6,7 @@
 config MLXSW_CORE
 	tristate "Mellanox Technologies Switch ASICs support"
 	select NET_DEVLINK
-	select MLXFW
+	imply MLXFW
 	select AUXILIARY_BUS
 	help
 	  This driver supports Mellanox Technologies Switch ASICs family.
-- 
2.8.4

